BUILD.bazel

```
@@ -54,9 +66,45 @@ cc_library(
 	visibility = ["//:__subpackages__"],
 )
 
+apple_binary(
+	name = "SDL2_mac",
+	binary_type = "dylib",
+	deps = [
+		":SDL2_lib",
+	],
+	platform_type = "macos",
+	visibility = ["//:__subpackages__"],
+)
+
+genrule(
+   name = "SDL2_dylib",
+   srcs = ["SDL2_mac"],
+   outs = ["SDL2.dylib"],
+   cmd = "cp $(location SDL2_mac) $(location SDL2.dylib)",
+)
+
+cc_binary(
+	name = "libSDL2.so",
+	linkshared = True,
+	deps = [
+		":SDL2_lib",
+	],
+	visibility = ["//visibility:public"],
+)
+
+cc_binary(
+	name = "SDL2.dll",
+	linkshared = True,
+	deps = [
+		":SDL2_lib",
+	],
+	linkopts = SDL_LINKOPTS,
+	visibility = ["//visibility:public"],
+)
+
 cc_library(
-	name = "SDL2",
-	hdrs = glob(["include/**/*.h"]),
+	name = "SDL2_lib",
```

Is it possible to leave this target name as "SDL2"? A bit of a nicer name for anyone who is using SDL2 statically.
```suggestion
	name = "SDL2",
```

---

If it's named "SDL2", then the windows build of "SDL2.dll" fails (naming conflict on bazel).
One of the two has to be renamed (or a genrule has to be created, or something else), and I just chose the static one, but I'm open to suggestions.

---

Ah that's unfortunate. With that constraint `SDL2_lib` is a decent name then!